WordPress/openverse
# Fix release contributions by removing explicit author link creation

## Fixes



Fixes an issue wherein contributors were no longer linked on a GitHub release page, see https://github.com/WordPress/openverse/releases/tag/api-2023.05.15.21.51.49 vs https://github.com/WordPress/openverse/releases/tag/ingestion_server-2023.05.03.22.29.52

## Description



In #2036, I removed the explicit "credits" section from the release/changelog generation in hopes that GitHub's automatic "Contributors" section would suffice. Unfortunately, in that PR I also pre-rendered the GitHub usernames as links (e.g. `[@$AUTHOR](https://github.com/$AUTHOR)` rather than just `$AUTHOR`). This meant that GitHub was no longer doing it's automatic mentioning behind the scenes, and so it could no longer create the "Contributors" section automatically. Per step 8 of their release document creation docs[^1] _(emphasis mine)_:

> In the "Describe this release" field, type a description for your release. **If you @mention anyone in the description, the published release will include a Contributors section with an avatar list of all the mentioned users.** Alternatively, you can automatically generate your release notes by clicking Generate release notes.

I believe that by making those Markdown links as part of the generation, I was circumventing GitHub's mechanism for doing this linking.

[^1]: https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release

As such, I've removed the changelog exclusion from the `link_usernames.py` script and also removed all the explicit links from our already existing changelog so as not to have the rudimentary parser I wrote double up on links (I used the regex pattern `\[(@[A-Za-z0-9-]+)\].*?\)` and replacement `$1` in PyCharm to accomplish this).

This should also now make future releases appropriately include the automatic "Contributors" section.

## Testing Instructions


Run `just documentation/live` locally and check that each of the changelogs still appropriately link the GitHub usernames (alternatively, wait for our GitHub actions to render the docs site and use the link below).

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
